test_runner/regress/test_pageserver_generations.py

```diff
@@ -273,12 +277,15 @@ def test_deferred_deletion(neon_env_builder: NeonEnvBuilder):
 
 
 @pytest.mark.parametrize("keep_attachment", [True, False])
+@pytest.mark.parametrize("validate_before", [True, False])
 def test_deletion_queue_recovery(
-    neon_env_builder: NeonEnvBuilder, pg_bin: PgBin, keep_attachment: bool
+    neon_env_builder: NeonEnvBuilder, pg_bin: PgBin, keep_attachment: bool, validate_before: bool
 ):
     """
     :param keep_attachment: If true, we re-attach after restart.  Else, we act as if some other
     node took the attachment while we were restarting.
+    :param validate_before: If true, we wait for deletions to be validated before restart.  This
+    makes the elegible to be executed after restart, if the same node keeps the attachment.
```

Comment:
```suggestion
    makes them elegible to be executed after restart, if the same node keeps the attachment.
```
I suppose you meant this?